Update konflux references and hermetic build

rakshithakamath94 opened the pull request.

Commit:
Update konflux references and hermetic build

Signed-off-by: Rakshitha-Kamath <Rakshitha.Kamath@ibm.com>

andrewschoen reviewed (changes requested):
A couple things I see here. For 8.1 we'll want to use ubi9 based containers, not rhel10. Also, we should be using rhsm-pulp.corp URLs instead of cdn.redhat. Thanks!

rakshithakamath94 commented:
Hi @andrewschoen ,

The repo file was wrong. In the rpms.in.yaml I was using `rpm.repo` file which was pointing to pulp repos. I have removed `redhead.repo` and added `rpms.repo` file.

rakshithakamath94 commented:
@andrewschoen Thanks for the review.
I have addressed the review comments and I am merging this PR.
